aequilibrae/paths/select_link.py

```diff
@@ -0,0 +1,164 @@
+import os
+from collections import defaultdict
+from typing import List, Dict
+import numpy as np
+import pandas as pd
+
+from aequilibrae.paths.assignment_paths import AssignmentPaths, TrafficClassIdentifier
+from aequilibrae import logger
+
+
+# TODO:
+#  - implement per-iteration contribution for CFW and BFW
+#  - some assertions as per constructor comments
+#  - demand matrix operations do not take cores into account (i.e. we assume we have a 2d demand matrix per class)
+#  - tests
+
```

Comment:
We could add a TODO here for later that is to have the code to find the matrices that were used from the assignment report.

Reply:
yes I'd like that. at the moment we're passing in the demand, but doing this automatically would be much better